### ISSUE TITLE
Search for docs site

### ISSUE THREAD

**Issue by armanbilge** [typelevel/sbt-typelevel#267]
I discussed this once with @valencik. The rough idea would be to generate a search index with Apache Lucene and feed it into lunr.js. Or something ...

See also:
- https://github.com/http4s/http4s/issues/5457

**Comment by Kevin-Lee** [comment]
I think this is a really important feature for documents.
If there's no progress on this issue, could you please consider using [DocSearch](https://docsearch.algolia.com/)?

**Comment by armanbilge** [comment]
@Kevin-Lee thanks, I also think this is very important.

Sure, how would a DocSearch integration work? It seems to me that each project would need to apply independently?

**Comment by Kevin-Lee** [comment]
@armanbilge Yes, you're right. As far as I know, each project should have its own application separately. However, once application is submitted then it requires to reply to one confirmation email from DocSearch and that's it.

**Comment by armanbilge** [comment]
Cool! That sounds pretty straightforward :)

Does anyone want to experiment with adding DocSearch to a Laika site? So we can get an idea of what the integration is like?

**Comment by Kevin-Lee** [comment]
@armanbilge To clarify, it's an application per a doc site.
So for example,
For Cats,
DOCUMENTATION OR BLOG URL: `https://typelevel.org/cats`
REPOSITORY URL: `https://github.com/typelevel/cats`

Once connected, you can monitor it through https://www.algolia.com and you will also have access to its [Crawler](https://crawler.algolia.com/). So I recommend you use one email address to create a Algolia account if you want to monitor all TypeLevel doc sites. If each site has a different owner, you probably want to use the owner's email.

I also recommend reading [this page](https://docsearch.algolia.com/docs/who-can-apply/).

**Comment by armanbilge** [comment]
This is happening!!

https://cozydev-pink.github.io/protosearch/

With a demo: https://cozydev-pink.github.io/protosearch/searchdocs/index.html

**Comment by Kevin-Lee** [comment]
@armanbilge That's awesome! Thank you so much!
